src/elements/element.line.js

```diff
@@ -52,57 +33,104 @@ function setStyle(ctx, vm) {
 	ctx.strokeStyle = vm.borderColor;
 }
 
-function normalPath(ctx, points, spanGaps, vm) {
-	const steppedLine = vm.steppedLine;
-	const lineMethod = steppedLine ? helpers.canvas._steppedLineTo : helpers.canvas._bezierCurveTo;
-	let move = true;
-	let index, currentVM, previousVM;
+function lineTo(ctx, previous, target) {
+	ctx.lineTo(target.x, target.y);
+}
+
+function getLineMethod(line) {
+	if (line.steppedLine) {
+		return _steppedLineTo;
+	}
+
+	if (line.tension) {
+		return _bezierCurveTo;
+	}
+
+	return lineTo;
+}
+
+function getInterpolationMethod(line) {
+	if (line.steppedLine) {
+		return _steppedInterpolation;
+	}
 
-	for (index = 0; index < points.length; ++index) {
-		currentVM = points[index]._view;
+	if (line.tension) {
+		return _bezierInterpolation;
+	}
 
-		if (currentVM.skip) {
-			move = move || !spanGaps;
+	return _pointInLine;
+}
+
+function pathSegment(ctx, line, segment, options) {
+	const {start, end, loop} = segment;
+	const {points, _view: vm} = line;
+	const lineMethod = getLineMethod(vm);
```

Comment:
You could probably call `getLineMethod` just one per line instead of once per segment

Reply:
Filler is drawing segments separately. This could be resolved and cached internally, but would have to detect changes in options then. I don't think its worth the trouble.